Do not require wheel for building

hroncok opened the pull request:
##### SUMMARY

- current version of setuptools (70.1+) does not need wheel at all
- older versions of setuptools would depend on wheel when building wheels (but not sdists)

<!--- Describe the change below, including rationale and design decisions -->

<!--- Add "Fixes #1234" or steps to reproduce the problem if there is no corresponding issue -->

##### ISSUE TYPE

<!--- Pick one below and delete the rest -->

- Bugfix Pull Request
- Feature Pull Request

(Not sure which one, this is neither; it's just cleaning.)

hroncok committed:
Do not require wheel for building


- current version of setuptools (70.1+) does not need wheel at all
- older versions of setuptools would fetch wheel when building wheels (but not sdists)

mattclay commented on pyproject.toml:
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools >= 66.1.0, <= 80.3.1", "wheel == 0.45.1"]  # lower bound to support controller Python versions, upper bound for latest version tested at release
+requires = ["setuptools >= 66.1.0, <= 80.3.1"]  # lower bound to support controller Python versions, upper bound for latest version tested at release
```
If we're going to drop the wheel pin, we'll need to raise our minimum setuptools version to 70.1.0 to ensure the wheel package isn't needed.

hroncok replied:
I was wondering if there could be a requirement constraint in here, but probably not.

Anyway, would raising the minimum setuptools version be acceptable?

mattclay replied:
Given that setuptools 70.1.0 was released over a year ago, I don't think there should be any issues using that as the new minimum.

Worst case, if a package maintainer does need to use an older setuptools, they'll have to deal with any potential incompatibilities themselves -- which could happen since we'll no longer be testing our builds against a lower version.

webknjaz replied:
FTR, old `setuptools` does depend on `wheel` through PEP 517.

hroncok replied:
Yes. And when that happens, we cannot easily restrict the wheel version.

webknjaz replied:
Technically, using `PIP_CONTRAINT` is easy enough. The file for it can be generated with `pip-tools`, for example. pip 25.3 will finally have first-class support for this via `--build-constraint` thanks to the recently-merged https://github.com/pypa/pip/pull/13534. And the https://github.com/pypa/build/issues/292 discussion is being revived in relation to that too.

hroncok commented on pyproject.toml:
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools >= 66.1.0, <= 80.3.1", "wheel == 0.45.1"]  # lower bound to support controller Python versions, upper bound for latest version tested at release
+requires = ["setuptools >= 66.1.0, <= 80.3.1"]  # lower bound to support controller Python versions, upper bound for latest version tested at release
```
```suggestion
requires = ["setuptools >= 70.1, <= 80.3.1"]  # lower bound to support controller Python versions, upper bound for latest version tested at release
```

hroncok committed:
Pin setuptools to a version not requiring wheel

Now when we don't list wheel,
we are unable to pin it to a particular version.

Instead, use setuptools version that no longer uses it.

mattclay commented:
/azp run
